packages/shogi-bot/src/bot.js

```diff
@@ -14,16 +14,24 @@ export default class ShogiBot extends LogBot {
     super(config, view);
   }
 
+  /**
+   * @param {View} view
+   */
   async handleUpdate(view) {
     const game = new Shogi();
-    view.log.forEach(({ from, to }) => game.move({ from, to }));
-    console.log(game.ascii());
+    view.log.forEach(message => game.applyMessage(message));
+
+    console.log(game.state.ascii());
 
-    const moves = game.getMoves();
+    // TODO(burdon): Consider drops.
+    const moves = game.state.getMoves();
     if (moves.length) {
       const move = moves[Math.floor(Math.random() * moves.length)];
-      const { from, to } = game.move(move);
-      view.appendChange({ from, to });
+
+      const message = game.createMessage(move);
```

Comment:
In discussions around chess repo & chess-bot state machine we concluded that game state machine should be modified by only applying logs (no direct modifications)